### PR TITLE
Add new remote application watchers plus new tests

### PR DESCRIPTION
**Pull request by wallyworld**
Add / replace new watchers:
- WatchRemoteApplicationRelations
- WatchLocalRelationUnits

New tests are also added for existing cmr code which was not yet tested. Some existing watcher code is currently unused but is retained until all refactoring is completed.

QA: code is currently unused, so just a quick bootstrap with lxd

### CODE REVIEW

File: api/remoterelations/remoterelations.go [hunk 1]
```diff
@@ -67,6 +67,32 @@ func (st *State) WatchRemoteApplicationRelations(application string) (watcher.Re
...
+		return nil, errors.NotValidf("relation key %q", relationKey)
+	}
+	relationTag := names.NewRelationTag(relationKey)
+	args := params.Entities{
+		Entities: []params.Entity{{Tag: relationTag.String()}},
+	}
+	var results params.RelationUnitsWatchResults
+	err := st.facade.FacadeCall("WatchLocalRelationUnits", args, &results)
+	if err != nil {
+		return nil, err
```
Comment (perrito666) [inline api/remoterelations/remoterelations.go]: errors.Trace(err) ?

File: apiserver/remoterelations/remoterelations_test.go [hunk 2]
```diff
@@ -177,15 +211,16 @@ func (s *remoteRelationsSuite) TestWatchRemoteApplicationRelationRemovedRace(c *
...
+		Change: &params.RemoteRelationsChange{},
 	}})
 	s.st.CheckCalls(c, []testing.StubCall{
 		{"WatchRemoteApplicationRelations", []interface{}{"db2"}},
 		{"KeyRelation", []interface{}{"db2:db django:db"}},
 	})
 }
 
-func (s *remoteRelationsSuite) TestWatchRemoteApplicationRelationUnitRemoved(c *gc.C) {
+// TODO(wallyworld) - underlying code not currently used
```
Comment (perrito666) [inline apiserver/remoterelations/remoterelations_test.go]: Why the code not being used is reason to not test it? is the test not yet functional?

Comment (wallyworld) [inline apiserver/remoterelations/remoterelations_test.go]: This whole section of code will be refactored - the code is essentially 12 months old and several branches are required to get things in shape. All this code is hidden behind a feature flag.

File: api/remoterelations/remoterelations.go (hunk 1 above)
Comment (wallyworld) [inline api/remoterelations/remoterelations.go]: fixed

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**